### PR TITLE
Request app exit instead of SIGTERM (darwin)

### PR DESCRIPTION
**Pull request by gabriel**
Requests app exit before doing a SIGKILL. I tested both scenarios locally but needs a few days of real testing.

### CODE REVIEW

**Comment by gabriel** [comment]
cc @keybase/updater-hackers

**Comment by gabriel** [comment]
OK this is ready to review

**Comment by chrisnojima** [comment]
seems fine to me, can we get some go 👀  maybe @jzila ?

File: keybase/platform_darwin.go [hunk 1]
```diff
@@ -151,8 +151,16 @@ func (c context) restart(wait time.Duration, delay time.Duration) error {
...
 
-	c.log.Infof("Terminating %s", appProcPath)
-	process.TerminateAll(process.NewMatcher(appProcPath, process.PathContains, c.log), time.Second, c.log)
+	// Request the app to exit (instead of sending a SIGTERM)
+	if appExitResult, err := command.Exec(c.config.keybasePath(), []string{"ctl", "app-exit"}, 5*time.Second, c.log); err != nil {
+		c.log.Warningf("Error requesting app exit: %s (%s)", err, appExitResult.CombinedOutput())
+	}
+	// Give the app time to exit
+	time.Sleep(time.Second)
+	// SIGKILL the app if it failed to exit
```
Comment (jzila) [inline keybase/platform_darwin.go]: why `KILL` and not `TERM`?

**Comment by jzila** [comment]
LGTM except for one question

**Comment by gabriel** [comment]
We do KILL because it we're changing the Keybase.app to do `ctl stop` on SIGTERM, which stops the world (and on updates we want just the app to quit for restart). See https://keybase.atlassian.net/browse/DESKTOP-1552 for more info. There was a long discussion about this topic BTW and this was the consensus.

**Pull request merged**